nixio/pycore/h5dataset.py

```diff
@@ -43,6 +43,8 @@ def write_data(self, data, count=None, offset=None):
             self.dataset[:] = data
 
     def read_data(self, data, count=None, offset=None):
+        if 0 in self.dataset.shape or len(self.dataset.shape) == 0:
+            return
```

Comment:
Well, that's exactly what it does now. The `data` object is supposed to hold the data. In the standard case, the function ends with `self.dataset.read_direct(data)` and nothing is returned.
So when we do a simple `return`, the `data` array is just returned empty.